code/modules/reagents/Chemistry-Machinery.dm

```diff
@@ -1098,6 +1098,22 @@ USE THIS CHEMISTRY DISPENSER FOR MAPS SO THEY START AT 100 ENERGY
 		src.updateUsrDialog()
 		return 0
 
+	//Fill the machine with the pill collector!
+	if(istype(O, /obj/item/weapon/storage/bag/chem))

```

Comment:
This is almost entirely copied from the previous section. If they really are so similar you can just use istype(pill collector) || istype(plant bag).